app/controllers/status_controller.rb

```diff
@@ -0,0 +1,49 @@
+### This will go away. The logic in here would go into the Inisghts Common Status Controller
+class StatusController < ActionController::API
+  def health
+    @output = {}
+
+    if db_down?
+      @output[:status] = "fail"
+      @output[:database_status] = "down"
+    else
+      @output[:status] = migration_context.needs_migration? "warn" : "pass"
+      @output[:database_status] = "up"
+    end
+
+    @output[:pending_migrations] = pending_migrations
+
+    render :json => @output, :status => status
+  end
+
+  private
+
+  def migration_context
+    ActiveRecord::Base.connection.migration_context
+  end
+
+  def pending_migrations
+    migration_context.migrations_status.collect(&:first).count("down")
+  end
+
+  def status
+    case @output[:status]
+    when "fail"
+      :internal_server_error
+    when "warn"
+      :partial_content
+    else
+      :ok
+    end
+  end
+
+  def db_down?
+    ActiveRecord::Base.connection
+    false
+  rescue StandardError
+    true
+  end
+end
+
+###### The above will change to this:
+# class StatusController < Insights::API::Common::StatusController; end
```

Comment:
Basically this entire class would live in the common gem that way in order to pull in the behavior the only thing required would be this line here in the `status_controller.rb` file as well as the route configuration.